AmplifyCombineSupport/AmplifyCombineSupportTests/DataStoreCombineSupportChainTests.swift

```diff
@@ -0,0 +1,236 @@
+        let receivedError = expectation(description: "Received error")
+        receivedError.isInverted = true
+
+        let responder: SaveResponder = { model, _ in .success(model) }
+        plugin.responders.save = responder
+
+        let sink = Publishers.Zip(
+            Amplify.DataStore.save(Person(name: "Peter Parker (Caller)")),
+            Amplify.DataStore.save(Person(name: "Mary Jane (Callee)"))
+        ).flatMap { caller, callee in
+            Amplify.DataStore.save(PhoneCall(caller: caller, callee: callee))
+        }.flatMap { phoneCall in
+            Publishers.Zip(
+                Amplify.DataStore.save(
+                    Transcript(text: "call transcript in english",
+                               language: "en",
+                               phoneCall: phoneCall)
+                ),
+                Amplify.DataStore.save(
+                    Transcript(text: "transcripción de la llamada en español",
+                               language: "es",
+                               phoneCall: phoneCall)
+                )
+            )
+        }.sink(receiveCompletion: { completion in
+            if case .failure = completion {
+                receivedError.fulfill()
+            }
+        }, receiveValue: { _ in receivedValue.fulfill() })
+
+        waitForExpectations(timeout: 0.05)
+        sink.cancel()
+    }
+
+    func testChainedOperationsFail() {
+        let receivedValue = expectation(description: "Received value")
+        receivedValue.isInverted = true
+        let receivedError = expectation(description: "Received error")
+
+        // Craft the responder so it fails in the middle of the chain
+        let responder: SaveResponder = { model, _ in
+            switch model {
+            case is PhoneCall:
+                return .failure(DataStoreError.invalidModelName("Blah blah"))
+            default:
+                return .success(model)
+            }
+        }
+
+        plugin.responders.save = responder
+
+        let sink = Publishers.Zip(
+            Amplify.DataStore.save(Person(name: "Peter Parker (Caller)")),
+            Amplify.DataStore.save(Person(name: "Mary Jane (Callee)"))
+        ).flatMap { caller, callee in
+            Amplify.DataStore.save(PhoneCall(caller: caller, callee: callee))
+        }.flatMap { phoneCall in
+            Publishers.Zip(
+                Amplify.DataStore.save(
+                    Transcript(text: "call transcript in english",
+                               language: "en",
+                               phoneCall: phoneCall)
+                ),
+                Amplify.DataStore.save(
+                    Transcript(text: "transcripción de la llamada en español",
+                               language: "es",
+                               phoneCall: phoneCall)
+                )
+            )
+        }.sink(receiveCompletion: { completion in
+            if case .failure = completion {
+                receivedError.fulfill()
```

Comment:
is there a way to know that it was when saving the PhoneCall that failed?

Reply:
Not unless the failing operation enhances the error in some way that makes it obvious. In that way, this pattern is similar to a `do/catch` block, or a Promise chain in the async world--error handling is consolidated in one place.